src/test/netbase_tests.cpp

```diff
@@ -109,6 +109,11 @@ BOOST_AUTO_TEST_CASE(subnet_test)
     BOOST_CHECK(CSubNet("1.2.3.0/24") != CSubNet("1.2.4.0/255.255.255.0"));
     BOOST_CHECK(CSubNet("1.2.3.0/24").Match(CNetAddr("1.2.3.4")));
     BOOST_CHECK(!CSubNet("1.2.2.0/24").Match(CNetAddr("1.2.3.4")));
+
+    BOOST_CHECK(CSubNet("1.2.2.1/24").Match(CNetAddr("1.2.2.4")));
```

Comment:
So to be clear: these tests fail without this change?
I doubt you need as many changes to fix this. Particularly the change to `match` appears unnecessarily complex. It is not simply a matter of bitwise AND-ing the `addr` with `netmask` after parsing?